common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -53,9 +56,13 @@
     // that the caller only requests each chunk one at a time, in order.
     int curChunk = 0;
 
+    // Used to keep track of the number of chunks being transferred and not finished yet.
+    AtomicLong chunksBeingTransferred;
```

Comment:
Really? we do `chunksBeingTransferred++` and `chunksBeingTransferred--`? I think it's not safe.